README.md

```diff
@@ -78,12 +146,153 @@ val records: DataFrame = sqlContext.redshiftFile(path, Seq("name", "age"))
+
+<table>
+ <tr>
+    <th>Parameter</th>
+    <th>Required</th>
+    <th>Default</th>
+    <th>Notes</th>
+ </tr>
+
+ <tr>
+    <td><tt>redshifttable</tt></td>
+    <td>Yes</td>
+    <td>No default</td>
+    <td>The table to create or read from in Redshift</td>
+ </tr
+ <tr>
+    <td><tt>jdbcurl</tt></td>
+    <td>Yes</td>
+    <td>No default</td>
+    <td>
+<p>A JDBC URL, of the format, <tt>jdbc:subprotocol://host:port/database?user=username&password=password</tt></p>
 
-Some breaking changes were made in version 0.3. Users should make the following changes in their 
-code if they would like to use the 0.3 version:
+<ul>
+ <li><tt>subprotocol</tt> can be <tt>postgresql</tt> or <tt>redshift</tt>, depending on which JDBC driver 
+    you have loaded. Note however that one Redshift-compatible driver must be on the classpath and match 
+    this URL.</li>
+ <li><tt>host</tt> and <tt>port</tt> should point to the Redshift master node, so security groups and/or VPC will
+need to be configured to allow access from your driver application.
+ <li><tt>database</tt> identifies a Redshift database name</li>
+ <li><tt>user</tt> and <tt>password</tt> are credentials to access the database, which must be embedded
+    in this URL for JDBC, and your user account should have necessary privileges for the table being referenced. </li>
+    </td>
+ </tr>
+ <tr>
+    <td><tt>aws_access_key_id</tt></td>
+    <td>No, unless also unavailable from environment</td>
+    <td><a href="http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/credentials.html">Default Provider Chain</a></td>
+    <td>AWS access key, must have write permissions to the S3 bucket.</td>
+ </tr>
+ <tr>
+    <td><tt>aws_secret_access_key</tt></td>
+    <td>No, unless also unavailable from environment</td>
+    <td><a href="http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/credentials.html">Default Provider Chain</a></td>
+    <td>AWS secret access key corresponding to provided access key.</td>
+ </tr>
+ <tr>
+    <td><tt>tempdir</tt></td>
+    <td>Yes</td>
+    <td>No default</td>
+    <td>A writeable location in Amazon S3, to be used for unloaded data when reading and Avro data to be loaded into
+Redshift when writing. If you're using `spark-redshift` as part of a regular ETL pipeline, it can be useful to
+set a <a href="http://docs.aws.amazon.com/AmazonS3/latest/dev/object-lifecycle-mgmt.html">Lifecycle Policy</a> on a bucket
+and use that as a temp location for this data.
+    </td>
+ </tr>
+ <tr>
+    <td><tt>jdbcdriver</tt></td>
+    <td>No</td>
+    <td><tt>org.postgresql.Driver</tt></td>
+    <td>The class name of the JDBC driver to load before JDBC operations. Must be on classpath.</td>
+ </tr>
+ <tr>
+    <td><tt>overwrite</tt></td>
+    <td>No</td>
+    <td><tt>false</tt></td>
+    <td>If true, drop any existing data before writing new content. See also <tt>usestagingtable</tt></td>
```

Comment:
This is so that users can specify overwrite/append when using the `saveAsRedshiftTable` function in `package.scala`. I wasn't sure whether that public interface is really enough needed now that we're using data sources, but I put it there to match the existing input format API. For now, I just updated the docs to make it clear what this param is for, but we could potentially make this functionality data sources only if you prefer.